src/components/application_manager/rpc_plugins/app_service_rpc_plugin/src/commands/hmi/as_publish_app_service_request.cc

```diff
@@ -63,13 +63,23 @@ void ASPublishAppServiceRequest::Run() {
   smart_objects::SmartObject service_record =
       application_manager_.GetAppServiceManager().PublishAppService(manifest,
                                                                     false);
+  if (smart_objects::SmartType_Map != service_record.getType()) {
```

Comment:
Eh, either one works for this case due to the description for `PublishAppService`. I might argue that this covers more potential cases, as we really shouldn't be sending back this value if it is not a map. That said, I don't have significant feelings either way, so if you still think it should be changed, I can do so.

Reply:
From the function brief, In the case that the request failed it should send a `Null SmartObject value`. I think `empty` might be closer to the mark rather than checking the SmartType. That said if you'd prefer to use `NULL` you could change the return type to a Smart Object pointer

Reply:
I'd love to do that, but sadly that would be a major API change if I modified the return value, this is easier for now